<?xml version="1.0"?>
<launch>

	<!-- Collecting waypoints -->	
	<arg name="IS_CP" default="false"/>

	<!-- Desired forward speed-->
	<arg name="DESIRED_SPEED" default="3.0"/>

	<!--circlefixed_example.dat is a circle of radius 20m for mkz sim -->
	<!--ENTER /odom_waypoints.dat or /config/circlefixedutm_example.dat -->
	<arg name="WAYPOINTS_FILE" default="/odom_waypoints.dat "/>

	<!--:= IS_CP == TRUE for waypoint generation, IS_CP == FALSE for controller run -->
	<group unless="$(var IS_CP)">
		<node pkg="vehicle_control_mkz" exec="odompubtest_mkz.py" name="odom_publisher_node"/>
		<node pkg="vehicle_control_mkz" exec="long_control.py" name="long_control_node" output="screen">
			<param name="DESIRED_SPEED" value="$(var DESIRED_SPEED)"/>
		</node>
		<!--node pkg="vehicle_control_mkz" exec="lat_control.py" name="lat_control_node" output="screen">
			<param name="DESIRED_SPEED" value="$(var DESIRED_SPEED)"/>
			<param name="WAYPOINTS_FILE" value="$(var WAYPOINTS_FILE)"/>
		</node!-->
	</group>
		
	<group if="$(var IS_CP)">
		<node pkg="vehicle_control_mkz" exec="odompubtest_mkz.py" name="odom_publisher_node"/>
		<node pkg="vehicle_control_mkz" exec="collect_waypoints.py" name="waypoint_collection_node" output="screen">
			<param name="WAYPOINTS_FILE" value="$(var WAYPOINTS_FILE)"/>
		</node>
	</group>	
	
</launch>
